include/AAInterface.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "openauto/Service/IAndroidAutoInterface.hpp"
+#include "app/action.hpp"
+
+
+
+class AAInterface : public QObject, public openauto::service::IAndroidAutoInterface {
+    Q_OBJECT
+
+   public:
+    AAInterface();
+    ~AAInterface();
+    void mediaPlaybackUpdate(const aasdk::proto::messages::MediaInfoChannelPlaybackData& playback) override;
+    void mediaMetadataUpdate(const aasdk::proto::messages::MediaInfoChannelMetadataData& metadata) override;
+    void navigationStatusUpdate(const aasdk::proto::messages::NavigationStatus& navStatus) override;
+    void navigationTurnEvent(const aasdk::proto::messages::NavigationTurnEvent& turnEvent) override;
+    void navigationDistanceEvent(const aasdk::proto::messages::NavigationDistanceEvent& distanceEvent) override;
+    void injectButtonPressHelper(aasdk::proto::enums::ButtonCode::Enum buttonCode, Action::ActionState actionState);
+    static AAInterface *get_instance();
```

Comment:
id prefer not to introduce any more singletons

im thinking adding Session::AndroidAuto would be a nice way to give everyone access to this interface (via the arbiter) and to keep all things AA related together (like later moving the AA worker there as well)

Reply:
Let me play with this implementation and report back - I was trying to match existing code style (singletons), but if we want to move away from that to a Session namespace, that works for me.

Reply:
oh yeah singletons was how everyone was in the beginning

but now I think the only singleton left is the old Config class (which I still need to get rid of 😅)

Reply:
take a look in include/app/session.hpp and include/app/arbiter.hpp to see how its done now

Reply:
I think there's another one for the canbus stuff, but yea I'll take a look.

Reply:
oh shoot might have missed that in my refactor that, ill fix that soon™️